connector_magento/components/backend_adapter.py

```diff
@@ -153,9 +153,20 @@ def call(self, method, arguments, http_method=None, storeview=None):
             # record(method, arguments, result)
             return result
         except (socket.gaierror, socket.error, socket.timeout) as err:
-            raise NetworkRetryableError(
-                'A network error caused the failure of the job: '
-                '%s' % err)
+            if err.response is not None:
+                if err.response.headers.get('Content-Type') == 'application/json':
+                    error_text = err.response.json()
+                else:
+                    error_text = err.response.text
+                raise NetworkRetryableError(
```

Comment:
You have it the wrong way around. If there is a json, there was no network error.
So in this case, raise a different exception type. I would define a custom exception type `MagentoError` or something that you can pass the json to, so that it is available when we catch this particular error (we will use that in the follow-up tickets that depend on this one).

Reply:
I have made a MagentoError exception with json